### PR TITLE
Filtering out "soft deleted" entries.

### LINKED ISSUE #273: Filter out "soft" deleted entries?
### Pacifica Metadata version
0.12.3.1

### Pacifica Core Software versions
1.0

### Scenario:
I need to be able to retrieve entries from the rest API, but want to exclude anything that has been "soft deleted" (deleted column is not null). When pulling info for the EUS sync script, I would like to be able to just see things that aren't deleted.

### Steps to Reproduce:
I can't see any way to filter out soft deleted records from the REST endpoint

### PR DESCRIPTION
**Pull request by kauberry**
### Description

Minor addition to orm/base.py to enable the system to filter out soft deleted entries (entries with a non-null deleted date). 

This just "makes sense" to me. The system shouldn't show anything that has been "deleted" unless you specifically ask it to do so (by adding a query param of "include_deleted"

### Issues Resolved

Fixes #273 

### Check List

- [ ] All tests pass. See <https://github.com/pacifica/pacifica-docs/blob/master/development/TESTING.md>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kauberry** [comment]
Also not sure about the efficacy of this block of logic right beneath the new bits I added in base.py/where_clause...

```
        if 'deleted' in kwargs:
            if kwargs['deleted'] is None:
                where_clause &= Expression(
                    getattr(cls, 'deleted'), OP.IS, None)
            else:
                date_obj = datetime_converts(kwargs['deleted'])
                where_clause &= Expression(
                    getattr(cls, 'deleted'), OP.EQ, date_obj)
```
I can't see any way to get a "None" value to show up from a query arg. It always shows up as just an empty string ('')

**Comment by dmlb2000** [comment]
Well the truthy value of an empty string is false. So might just say, `if kwargs['deleted']:`.and see if that does what you want.

**Comment by kauberry** [comment]
@dmlb2000, I tried your suggestion locally, and that does indeed work as a filter. The bigger question that remains is in regards to what the default behavior should be. Should we, by default, be showing "deleted" items?

File: pacifica/metadata/orm/base.py [hunk 1]
```diff
@@ -209,14 +209,18 @@ def _date_operator_compare(date, kwargs, dt_converts=datetime_converts):
     def where_clause(cls, kwargs):
         """PeeWee specific extension meant to be passed to a PeeWee get or select."""
         where_clause = Expression(1, OP.EQ, 1)
+        # if 'include_deleted' not in kwargs and 'deleted' not in kwargs:
```
Comment (dmlb2000) [inline pacifica/metadata/orm/base.py:212]: These lines should be removed.

Comment (kauberry) [reply pacifica/metadata/orm/base.py]: Ok, I've removed those lines, but Travis seems to be complaining about code that I didn't change (not to my knowledge, anyhow)

Comment (kauberry) [reply pacifica/metadata/orm/base.py]: Again, not sure why pylint keeps puking.

**Comment by dmlb2000** [comment]
@kauberry could you please check out the changes, see if they make sense. This should still remove deleted objects by default. Hopefully the behavior is as follows:

Input: Not specifying 'deleted' at all
Behavior: filters out the items where deleted is set to something.

Input: Specifying 'deleted' as a key but the value has a truthy value of false (i.e. empty string, hash or list)
Behavior: Return items were the deleted may or may not be set

Input: Specifying 'deleted' as a key and the value has a truthy value of true
Behavior: Try converting the value to a datetime and use that to find deleted items

**Pull request merged**